tabs/tab_data.py

```
@@ -331,25 +332,45 @@ def _save_metadata():
+
+            # ✅ ใช้ DataGrid: รองรับข้อมูลจำนวนมากได้ดี (Virtual Scrolling) 
+            # Browser จะไม่ค้างเพราะโหลดเฉพาะแถวที่แสดงผล
+            return render.DataGrid(
+                d,
+                width="100%",
+                height="100%", # ให้ความสูงปรับตาม Card
+                filters=True   # แถม: เพิ่มช่อง Filter หัวคอลัมน์ให้ด้วย
+            )
+
+        except Exception as e:
+            logger.exception("Error rendering preview: %s", e)
```

Ruff (0.14.10)

Redundant exception object included in `logging.exception` call

(TRY401)